JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/CompileAndRunBean.java

```diff
@@ -1,27 +1,200 @@
+        sources = fileBean.getProjectFiles(projectHash);
+        String mes = "";
+        if (!getMessage().equals("")) {
+            Method mainMeth = findMain(sources, getClassLoader());
+            String[] arguments = {};
+            try {
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                PrintStream printStream = new PrintStream(baos);
+                PrintStream old = System.out;
+                System.setOut(printStream);
+                mainMeth.invoke(null, (Object) arguments);
+                System.out.flush();
+                System.setOut(old);
+                mes = baos.toString();
+            } catch (NullPointerException e) {
+                LOG.error(e.getMessage(), e);
+            } catch (InvocationTargetException e) {
+                LOG.error(e.getMessage(), e);
+            } catch (IllegalAccessException e) {
+                LOG.error(e.getMessage(), e);
+            }
+        }
+        return mes;
+    }
+
+    public Method findMain(List<File> sources, ClassLoader clazzLoader) {
+        int count = 0;
+        Class clazz = null;
+        Method mainMeth = null;
+        for (int i = 0; i < sources.size(); i++) {
+            File file = sources.get(i);
+            // typeId = 8 - "class", typeId = 9 - "interface", typeId = 10 - "exception"
+            // typeId = 5 - "runnable", typeId = 11 - "enum", typeId = 12 - "annotation"
+            int typeId = file.getType().getTypeId();
+            if (typeId == 8 || typeId == 9 || typeId == 10 || typeId == 5 || typeId == 11 || typeId == 12) {
+                String src = findSrc(sources);
+                String packSource = file.getPath().replaceFirst(src, "").replace("/", ".").replaceFirst(".", "");
+//                String pack = "";
+//                if (!packSource.equals("")) {
+//                    pack = packSource + ".";
+//                }
+                String className = packSource;//pack + file.getFileName();
+                try {
+                    clazz = clazzLoader.loadClass(className);
+                    Method[] meth = clazz.getDeclaredMethods();
+                    int j = 0;
+                    while (meth != null && j != meth.length) {
+                        if (meth[j].getName().equals("main")) {
+                            mainMeth = meth[j];
+                            count++;
+                        }
+                        j++;
+                    }
+                } catch (ClassNotFoundException e) {
+                    LOG.error("ClassNotFoundException", e);
+                } catch (NullPointerException e) {
+                    LOG.error("NullPointerException", e);
+                }
+            }
+        }
+
+        if (count != 1) {
+            try {
+                throw new Exception();
+            } catch (Exception e) {
+                LOG.error("Too many main functions", e);
+            }
+            return null;
```

Comment:
Holy crap! WHYYYY???
I'm trying to find proper words...

1) Throwing exceptions here and there is VERY BAD.
Exception is something EXCEPTIONAL, not normal. Your logic should avoid them where possible.
And throwing exceptions is for noticing a caller of this API, not for the same method.
2) You are muting information that is very important for a user: his/her Java class has multiple main methods, which is not allowed for sure. This should go onto UI, not into logs. How should a user guess why main is not found? Request system logs from administrator?
3) Your code should not duplicate the logic of Java Compiler. You should call it and push all its messages onto UI (need to be designed adequately). And then you can assume that the code your are compiling does not have any compilation errors, e.g. multiple public static void main(String[] args) methods.
4) Although, multiple public static void main(String[] args) within one class are not allowed for obvious reasons, you still can have method overloading for main. For that reason, you should check not only method name, but the args it accepts as well.